### ISSUE TITLE
Wrong terminal location when fullscreen application is focused

### ISSUE THREAD

**Issue by TheRealGramdalf** [ddterm/gnome-shell-extension-ddterm#239]
When using the terminal when another application is in full screen, the terminal starts from the lowest position of where the top bar would be. (see pictures)

Regular (non fullscreen)
![image](https://user-images.githubusercontent.com/79593869/201209187-368c3b8a-1c0e-4480-b9e7-1d32d680e443.png)

In fullscreen
![image](https://user-images.githubusercontent.com/79593869/201209296-459ee321-65fc-4880-bbcc-51195d43270f.png)

Ideally, the terminal would start from the top pixel of the screen if the focused application is in fullscreen (like in the second picture), but from beneath the top bar when the focused application is not (as in the first picture).

Some extra logic would be required, but that's the basic issue.

I tested this while disabling all other extensions, so I'm reasonably sure that it's not to do with another extension interfering. Distro is fedora 36 (gnome 42.4), using x11 (due to some issues with nvidia+wayland),

**Comment by amezin** [comment]
Yes, I understand and I am able to reproduce the issue.
I rarely use fullscreen mode - that's why this issue was unnoticed for so long.

**Comment by TheRealGramdalf** [comment]
Yeah, it's definitely not super common. I run an MC server so I'll have the game in fullscreen and use the terminal to look at logs at the same time. Kinda niche, so it's totally understandable as to why you didn't know. It's not super urgent, as it's purely aesthetic, so take your time! In the mean time, I'll continue to enjoy your amazing plugin.

**Comment by amezin** [comment]
Unfortunately, it seems that GNOME Shell doesn't allow moving windows to locations overlapping the panel. Even if there's a fullscreen window. In this case, the only option is to make ddterm fullscreen too

**Comment by amezin** [comment]
I have plans to write an xterm.js-based terminal emulator using GNOME Shell's UI (instead of the current implementation with a separate Gtk application). It should solve all window management issues, like this one.

However, it won't happen soon, as it will likely be a separate project, written (almost) from scratch.

https://github.com/ddterm/gnome-shell-extension-ddterm/issues/269

**Comment by alupula-vk** [comment]
Try using a tile instead of the full screen....check whether it works! Even I  use the full screen mode rarely!!
